include/alp/omp/matrix.hpp

```diff
@@ -52,41 +55,45 @@ namespace alp {
 			> * = nullptr
 		>
 		typename internal::new_container_type_from<
-			typename SourceMatrix::template view_type< view::cross_backend >::type
-		>::template change_backend< reference >::type
+			typename SourceMatrix::template view_type< view::gather >::type
+		>::template change_backend< config::default_sequential_backend >::type
 		get_view( SourceMatrix &source, const size_t tr, const size_t tc, const size_t rt, const size_t br, const size_t bc ) {
 
 			(void) rt;
 
-			using value_type = typename SourceMatrix::value_type;
-
 			// get the container
 			const auto &distribution = getAmf( source ).getDistribution();
-			const size_t thread_id = distribution.getThreadId( tr, tc );
-			value_type *const thread_buffer = internal::getBuffer( internal::getContainer( source ), thread_id );
-			value_type *const block_buffer = thread_buffer + distribution.getBlocksOffset( tr, tc, br, bc );
+			const size_t thread_id = tr * distribution.getThreadGridDims().second + tc;
+			const size_t block_id = br * distribution.getLocalBlockGridDims( tr, tc ).second + bc;
+			auto &container = internal::getLocalContainer( internal::getContainer( source ), thread_id, block_id );
 
 			// make an AMF
 			// note: When making a view over a vector, the second imf must be imf::Zero
+
+			// Type of AMF factory corresponding to the full local block's AMF
+			using original_amf_factory = alp::storage::AMFFactory< config::default_sequential_backend >::FromPolynomial<
```

Comment:
Do we need the factory or could we get the amf_type directly from the source type rather than recreating it?